homeassistant/components/trend/binary_sensor.py

```diff
@@ -194,6 +196,10 @@ def trend_sensor_state_listener(
             )
         )
 
+        if not (state := await self.async_get_last_state()):
+            return
+        self._state = state.state == STATE_ON
```

Comment:
The stored sensor state may be "unknown" which this code does not handle, maybe something like this will work:
```suggestion
        if state.state == STATE_UNKNOWN:
            return
        self._state = state.state == STATE_ON
```